frontend/packages/metal3-plugin/src/components/machine-cell.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { Icon } from 'patternfly-react';
+import { canHostAddMachine, DASHES, getHostMachineName } from 'kubevirt-web-ui-components';
+
+import { MachineModel } from '@console/internal/models';
+import { ResourceLink } from '@console/internal/components/utils';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+
+interface MachineCellProps {
+  host: K8sResourceKind
+}
+
+const MachineCell: React.FC<MachineCellProps> = ({ host }) => {
+  const machineName = getHostMachineName(host);
+
+  const {
+    metadata: { namespace },
+  } = host;
+
+  if (machineName) {
+    return (
+      <ResourceLink
+        kind={referenceForModel(MachineModel)}
+        name={machineName}
+        namespace={namespace}
+        title={machineName}
+      />
+    );
+  } else if (canHostAddMachine(host)) {
```

Comment:
Unnecessary `else` since you return inside the `if`

Reply:
Done